test/routes/add-article.test.ts

```diff
@@ -12,36 +14,42 @@ import createArticle from '../create-article';
 import runMiddleware from '../middleware';
 
 const makeRequest = async (
-  body: JsonLdObj = {},
+  dataset?: DatasetCore,
   next?: Next,
   articles: Articles = new InMemoryArticles(),
 ): Promise<Response> => (
-  runMiddleware(addArticle(), createContext({ articles, body }), next)
+  runMiddleware(addArticle(), createContext({ articles, dataset }), next)
 );
 
 describe('add article', (): void => {
   it('should return a successful response', async (): Promise<void> => {
     const articles = new InMemoryArticles();
-    const response = await makeRequest(createArticle(), undefined, articles);
+    const id = blankNode();
+    const name = literal('Article');
+    const response = await makeRequest(createArticle({ id, name }), undefined, articles);
 
     expect(response.status).toBe(201);
     expect(response.get('Location')).toBe('http://example.com/path-to/article-list');
     expect(await articles.count()).toBe(1);
-    expect((await all(articles))[0][1]['http://schema.org/name']).toEqual([{ '@value': 'Article' }]);
-  });
 
-  it('should throw an error if id is already set', async (): Promise<void> => {
-    const article = createArticle({ id: blankNode('12345') });
+    const [newId, dataset] = (await all(articles))[0];
 
-    await expect(makeRequest(article)).rejects.toBeInstanceOf(createHttpError.Forbidden);
-    await expect(makeRequest(article)).rejects.toHaveProperty('message', 'Article IDs must not be set (\'_:12345\' was given)');
+    expect(dataset.has(quad(newId, schema('name'), name))).toBe(true);
   });
 
   it('should throw an error if it is not a schema:Article', async (): Promise<void> => {
     const article = createArticle({ types: [schema.NewsArticle] });
 
     await expect(makeRequest(article)).rejects.toBeInstanceOf(createHttpError.BadRequest);
-    await expect(makeRequest(article)).rejects.toHaveProperty('message', 'Article type must be http://schema.org/Article (\'http://schema.org/NewsArticle\' was given)');
+    await expect(makeRequest(article)).rejects.toHaveProperty('message', 'No http://schema.org/Article found');
+  });
+
+  it('should throw an error if the articles does not have a blank node identifier', async (): Promise<void> => {
```

Comment:
just an additional test or also an additional behavior? Keeps things simple but is there an hidden decision for not allowing a more complex identifier (like a DOI)?

Reply:
(or maybe I'm confusing things)

Reply:
For now an article has a blank node identifier (ie it isn't dereferenceable). In the near future it will have its own identifier, but it will still need to be generated by the server.